client/src/pages/admin/pages/__tests__/add-data-source.test.tsx

```diff
@@ -0,0 +1,23 @@
+import { describe, expect, test } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import React from 'react';
+import { useAllProviders } from '@/mocks/setup-tests';
+import Form from '@/pages/admin/pages/add-data-source';
+
```

Comment:
think we should add test for success and failure cases, you should be able to do similar to what is done in the configuration test file :)

Reply:
Have just pushed this now! :)